Content.Server/Connection/Whitelist/WhitelistPrototype.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Threading.Tasks;
+using Robust.Shared.Network;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Connection.Whitelist;
+
+[Prototype("PlayerConnectionWhitelist")]
+public sealed class PlayerConnectionWhitelistPrototype : IPrototype
+{
+    [IdDataField]
+    public string ID { get; } = default!;
+
+    /// <summary>
+    /// Minimum number of players required for this whitelist to be active.
+    /// If there are less players than this, the whitelist will be ignored and the next one in the list will be used.
+    /// </summary>
+    [DataField]
+    public int MinimumPlayers { get; } = 0;
+
+    /// <summary>
+    /// Maximum number of players allowed for this whitelist to be active.
+    /// If there are more players than this, the whitelist will be ignored and the next one in the list will be used.
+    /// </summary>
+    [DataField]
+    public int MaximumPlayers { get; } = int.MaxValue;
+
+    [DataField]
+    public WhitelistCondition[] Conditions { get; } = default!;
+}
+
+public static class WhitelistExtensions
+{
+    public static async Task<(bool isWhitelisted, string? denyMessage)> IsWhitelisted(this PlayerConnectionWhitelistPrototype prototype, NetUserData data, ISawmill sawmill)
```

Comment:
Why is all this logic implemented as an extension method instead of in the only system that calls this?